### PR TITLE
ENH: improve structure of MNE-TFR code and isolate the single trial TFR computational step

### PR DESCRIPTION
**Pull request by dengemann**

### CODE REVIEW

**Comment by dengemann** [comment]
More cleaning up in preparaiton for future developments.

cc @agramfort @Eric89GXL

**Comment by dengemann** [comment]
some quick review would be nice. The code changes here will lay the fundament for future developments, i.e., alternative TFR estimators, functions to iterate over raw (not power or PLV) source TFRs, etc.

File: mne/minimum_norm/time_frequency.py [hunk 1]
```diff
@@ -208,21 +216,21 @@ def _compute_pow_plv(data, K, sel, Ws, source_ori, use_fft, Vh, with_plv,
 
                 if is_free_ori:
                     logger.debug('combining the current components...')
-                    sol = combine_xyz(sol, square=True)
-                else:
+                    sol = combine_xyz(sol, square=with_power)
+                elif with_power:
                     np.power(sol, 2, sol)
```
Comment (Eric89GXL) [inline mne/minimum_norm/time_frequency.py]: while you're in there, it should be faster to just say `sol *= sol` instead.

**Comment by Eric89GXL** [comment]
Looks reasonable to me. I think it's fine to iterate on the private functions as the use cases arise if need be, too.

**Comment by dengemann** [comment]
Thanks @Eric89GXL, not done yet. API is missing for returning single trials. Currently the innermost function is still accumulating.

**Comment by dengemann** [comment]
... the simplest would be a yield statement, depending on the outputmode. Either accumulate or `yield` the single trial TFR.

**Comment by agramfort** [comment]
ping us when ready to review.

**Comment by dengemann** [comment]
ping, we now have at the lowest level a function that computes the single trial TFR + PLV preparation.
upstream the functions should now look cleaner. A single trial TFR return function could now be easily conceivable as generator around `(_single_epoch_tfr(e, *args) fo e in epochs)`

**Comment by dengemann** [comment]
@Eric89GXL numpy seemlingly does more than just computing the power, with the python raw syntax you get nans. I keept it as it is now.

**Comment by dengemann** [comment]
I'm done for now. Can we merge this?

**Comment by Eric89GXL** [comment]
@dengemann I'm not sure if hiding `nan`s is a feature in this case. Can you investigate why they come up? It might indicate some bug with our code. It's probably better to explicitly mask the `nan` values, if that's what we want to do, than let a `numpy` function silently (and unknown to us up until now!) do it for us.

**Comment by Eric89GXL** [comment]
I'll open a separate issue because it isn't really part of your work here.

**Comment by Eric89GXL** [comment]
Thanks @dengemann, I assume you have some other PRs to follow?

**Comment by dengemann** [comment]
Yes, moving in small steps ;)

**Pull request merged**